data-prepper-plugins/otlp-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/otlp/http/OtlpHttpSender.java

```diff
@@ -0,0 +1,221 @@
+                final Request.Builder requestBuilder = new Request.Builder()
+                        .url(signedRequest.getUri().toString())
+                        .post(RequestBody.create(compressedPayload.get(), PROTOBUF))
+                        .addHeader("Content-Encoding", "gzip");
+
+                signedRequest.headers().forEach((key, values) -> {
+                    for (final String value : values) {
+                        requestBuilder.addHeader(key, value);
+                    }
+                });
+
+                final Request request = requestBuilder.build();
+
+                final long startTime = System.currentTimeMillis();
+                try (final Response response = httpClient.newCall(request).execute()) {
+                    final long duration = System.currentTimeMillis() - startTime;
+                    sinkMetrics.recordHttpLatency(duration);
+
+                    handleResponse(response);
+
+                    sinkMetrics.incrementPayloadSize(payload.length);
+                    sinkMetrics.incrementPayloadGzipSize(compressedPayload.get().length);
+                    return;
+                }
+            } catch (final IOException ioException) {
+                if (attempt < maxRetries) {
+                    final int jitter = random.nextInt(100);
+                    final int retryIndex = Math.min(attempt, retryDelaysMs.size() - 1);
+                    final int delay = retryDelaysMs.get(retryIndex) + jitter;
+                    try {
+                        sleeper.accept(delay);
+                        sinkMetrics.incrementRetriesCount();
+                    } catch (final RuntimeException runtimeException) {
+                        throw new IOException("Sender failed to sleep before retrying.", runtimeException);
+                    }
+                } else {
+                    throw new IOException("Max retries reached", ioException);
+                }
+            }
+        }
+    }
+
+    /**
+     * Handles the OTLP export response.
+     * Retries on 429, 502, 503, and 504 per OTEL spec. Logs other errors without retry.
+     * See: <a href="https://opentelemetry.io/docs/specs/otlp/#otlphttp-response">OTLP/HTTP Response</a>
+     *
+     * @param response The HTTP response
+     * @throws IOException For retryable errors
+     */
+    private void handleResponse(@Nonnull final Response response) throws IOException {
+        final int status = response.code();
+        sinkMetrics.recordResponseCode(status);
+
+        final byte[] responseBytes = response.body() != null
+                ? response.body().bytes()
+                : null;
+
+        if (status >= 200 && status < 300) {
+            handleSuccessfulResponse(responseBytes);
+            return;
+        }
+
+        final String responseBody = responseBytes != null ? new String(responseBytes, StandardCharsets.UTF_8) : "<no body>";
+        if (RETRYABLE_STATUS_CODES.contains(status)) {
```

Comment:
Here you need to inspect the `retry-after` header to be considered for retry backoff. Otherwise you are not implementing the protocol completely.